### ISSUE TITLE
Package tests are failing

### ISSUE THREAD

**Issue by thomas-moulard** [aws-ros-dev/cross_compile#16]
```
Starting >>> cross_compile
running pytest
running egg_info
writing /home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/PKG-INFO
writing dependency_links to /home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/dependency_links.txt
writing entry points to /home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/entry_points.txt
writing requirements to /home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/requires.txt
writing top-level names to /home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/top_level.txt
reading manifest file '/home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/SOURCES.txt'
writing manifest file '/home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/cross_compile.egg-info/SOURCES.txt'
running build_ext
============================= test session starts ==============================
platform linux -- Python 3.6.8, pytest-5.2.0, py-1.8.0, pluggy-0.13.0 -- /usr/bin/python3
cachedir: ros2_ws/build/cross_compile/.pytest_cache
rootdir: /home/ANT.AMAZON.COM/tmoulard, inifile: setup.cfg
plugins: ament-copyright-0.8.0, ament-pep257-0.8.0, ament-flake8-0.8.0, ament-lint-0.8.0, launch-testing-ros-0.9.1, launch-testing-0.9.1, ament-xmllint-0.8.0, ament-mypy-0.8.0, repeat-0.8.0, rerunfailures-7.0, cov-2.8.0
collecting ... collected 6 items

test/test_flake8.py::test_flake8 FAILED                                  [ 16%]
test/test_sysroot_compiler.py::test_get_workspace_image_tag PASSED       [ 33%]
test/test_sysroot_compiler.py::test_docker_config_args PASSED            [ 50%]
test/test_sysroot_compiler.py::test_sysroot_compiler_constructor PASSED  [ 66%]
test/test_sysroot_compiler.py::test_write_cc_build_setup_file FAILED     [ 83%]
test/test_sysroot_compiler.py::test_write_cc_system_setup_file FAILED    [100%]WARNING: Failed to generate report: No data to report.



=================================== FAILURES ===================================
_________________________________ test_flake8 __________________________________
test/test_flake8.py:61: in test_flake8
    assert not report.total_errors, \
E   AssertionError: flake8 reported 84 errors
E   assert not 74
E    +  where 74 = <flake8.api.legacy.Report object at 0x7f5abdd853c8>.total_errors
----------------------------- Captured stdout call -----------------------------

1     D102 Missing docstring in public method
1     D401 First line should be in imperative mood
50    E501 line too long (116 > 79 characters)
2     I100 Import statements are in the wrong order. 'import re' should be before 'from pathlib import Path'
7     Q000 Remove bad quotes
1     Q001 Remove bad quotes from multiline string
11    Q003 Change outer quotes to avoid escaping inner quotes
1     W293 blank line contains whitespace
1     D401 First line should be in imperative mood
6     E501 line too long (80 > 79 characters)
3     I100 Import statements are in the wrong order. 'from cross_compile.sysroot_compiler import DockerConfig' should be before 'import pytest' and in a different group.
----------------------------- Captured stderr call -----------------------------
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:20:1: I100 Import statements are in the wrong order. 'import re' should be before 'from pathlib import Path'
import re
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:23:1: I100 Import statements are in the wrong order. 'import subprocess' should be before 'from string import Template'
import subprocess
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:60:36: Q003 Change outer quotes to avoid escaping inner quotes
ROS_WS_DIR_ERROR_STRING = Template('\'$ros_ws\' does not exist in the sysroot directory. Make '
                                   ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:60:80: E501 line too long (95 > 79 characters)
ROS_WS_DIR_ERROR_STRING = Template('\'$ros_ws\' does not exist in the sysroot directory. Make '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:61:36: Q003 Change outer quotes to avoid escaping inner quotes
                                   'sure you copy your packages as \'$ros_ws/src\' into the '
                                   ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:61:80: E501 line too long (93 > 79 characters)
                                   'sure you copy your packages as \'$ros_ws/src\' into the '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:62:36: Q003 Change outer quotes to avoid escaping inner quotes
                                   '\'sysroot\' directory.')
                                   ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:63:34: Q003 Change outer quotes to avoid escaping inner quotes
QEMU_DIR_ERROR_STRING = Template('\'$qemu_dir\' does not exist in the sysroot directory. Make '
                                 ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:63:80: E501 line too long (95 > 79 characters)
QEMU_DIR_ERROR_STRING = Template('\'$qemu_dir\' does not exist in the sysroot directory. Make '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:64:34: Q003 Change outer quotes to avoid escaping inner quotes
                                 'sure you copy the binaries from \'/usr/bin/qemu-*\' into the '
                                 ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:64:80: E501 line too long (96 > 79 characters)
                                 'sure you copy the binaries from \'/usr/bin/qemu-*\' into the '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:66:36: Q003 Change outer quotes to avoid escaping inner quotes
QEMU_EMPTY_ERROR_STRING = Template('\'$qemu_dir\' is empty. Make sure you copy the binaries from '
                                   ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:66:80: E501 line too long (98 > 79 characters)
QEMU_EMPTY_ERROR_STRING = Template('\'$qemu_dir\' is empty. Make sure you copy the binaries from '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:67:36: Q003 Change outer quotes to avoid escaping inner quotes
                                   '\'/usr/bin/qemu-*\' into \'$qemu_dir\'.')
                                   ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:68:40: Q003 Change outer quotes to avoid escaping inner quotes
COPY_DOCKER_WS_ERROR_STRING = Template('Unable to copy the \'$dockerfile\' file. Make sure you '
                                       ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:68:80: E501 line too long (96 > 79 characters)
COPY_DOCKER_WS_ERROR_STRING = Template('Unable to copy the \'$dockerfile\' file. Make sure you '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:69:80: E501 line too long (90 > 79 characters)
                                       'have write permissions to the sysroot directory.')
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:70:43: Q003 Change outer quotes to avoid escaping inner quotes
SYSROOT_NOT_FOUND_ERROR_STRING = Template('Sysroot directory not found at \'$sysroot_dir\'. Make '
                                          ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:70:80: E501 line too long (98 > 79 characters)
SYSROOT_NOT_FOUND_ERROR_STRING = Template('Sysroot directory not found at \'$sysroot_dir\'. Make '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:71:80: E501 line too long (92 > 79 characters)
                                          'sure you specify the full path to the directory '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:72:43: Q003 Change outer quotes to avoid escaping inner quotes
                                          'containing \'sysroot\'.')
                                          ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:147:80: E501 line too long (98 > 79 characters)
    """A class that configures and builds a Docker container for cross-compiling ROS2 packages."""
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:156:80: E501 line too long (94 > 79 characters)
        """
        Construct a SysrootCompiler that builds a Docker container used for cross compilation.

        :param cc_root_dir: The directory containing the 'sysroot' directory with the ROS2
        workspace and QEMU binaries.
        :param ros_workspace_dir: The name of the directory containing the ROS2 packages (inside a
        'src' directory).
        :param platform: A custom object used to specify the the platform for cross-compilation.
        :param docker_config: A custom object used to specify the configuration of the Docker
        image to build.
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:158:80: E501 line too long (90 > 79 characters)
        """
        Construct a SysrootCompiler that builds a Docker container used for cross compilation.

        :param cc_root_dir: The directory containing the 'sysroot' directory with the ROS2
        workspace and QEMU binaries.
        :param ros_workspace_dir: The name of the directory containing the ROS2 packages (inside a
        'src' directory).
        :param platform: A custom object used to specify the the platform for cross-compilation.
        :param docker_config: A custom object used to specify the configuration of the Docker
        image to build.
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:160:80: E501 line too long (98 > 79 characters)
        """
        Construct a SysrootCompiler that builds a Docker container used for cross compilation.

        :param cc_root_dir: The directory containing the 'sysroot' directory with the ROS2
        workspace and QEMU binaries.
        :param ros_workspace_dir: The name of the directory containing the ROS2 packages (inside a
        'src' directory).
        :param platform: A custom object used to specify the the platform for cross-compilation.
        :param docker_config: A custom object used to specify the configuration of the Docker
        image to build.
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:162:80: E501 line too long (96 > 79 characters)
        """
        Construct a SysrootCompiler that builds a Docker container used for cross compilation.

        :param cc_root_dir: The directory containing the 'sysroot' directory with the ROS2
        workspace and QEMU binaries.
        :param ros_workspace_dir: The name of the directory containing the ROS2 packages (inside a
        'src' directory).
        :param platform: A custom object used to specify the the platform for cross-compilation.
        :param docker_config: A custom object used to specify the configuration of the Docker
        image to build.
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:163:80: E501 line too long (93 > 79 characters)
        """
        Construct a SysrootCompiler that builds a Docker container used for cross compilation.

        :param cc_root_dir: The directory containing the 'sysroot' directory with the ROS2
        workspace and QEMU binaries.
        :param ros_workspace_dir: The name of the directory containing the ROS2 packages (inside a
        'src' directory).
        :param platform: A custom object used to specify the the platform for cross-compilation.
        :param docker_config: A custom object used to specify the configuration of the Docker
        image to build.
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:167:29: Q000 Remove bad quotes
            raise TypeError("Argument `cc_root_dir` must be of type string.")
                            ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:169:29: Q000 Remove bad quotes
            raise TypeError("Argument `ros_workspace_dir` must be of type string.")
                            ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:169:80: E501 line too long (83 > 79 characters)
            raise TypeError("Argument `ros_workspace_dir` must be of type string.")
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:171:29: Q000 Remove bad quotes
            raise TypeError("Argument `platform` must be of type Platform.")
                            ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:173:29: Q000 Remove bad quotes
            raise TypeError("Argument `docker_config` must be of type DockerConfig.")
                            ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:173:80: E501 line too long (85 > 79 characters)
            raise TypeError("Argument `docker_config` must be of type DockerConfig.")
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:181:80: E501 line too long (83 > 79 characters)
        self._expected_dockerfile_directory = self._target_sysroot / DOCKER_WS_NAME
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:201:1: D401 First line should be in imperative mood
        """Checks to make sure the sysroot directory is setup correctly.
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:203:80: E501 line too long (94 > 79 characters)
        """Checks to make sure the sysroot directory is setup correctly.

        Raises FileNotFoundError's if any of the components necessary for cross compilation is
        missing. Copies the Dockerfile_workspace to the 'sysroot' directory in order to copy the
        assets to it (see https://docs.docker.com/engine/reference/builder/#copy).
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:204:80: E501 line too long (96 > 79 characters)
        """Checks to make sure the sysroot directory is setup correctly.

        Raises FileNotFoundError's if any of the components necessary for cross compilation is
        missing. Copies the Dockerfile_workspace to the 'sysroot' directory in order to copy the
        assets to it (see https://docs.docker.com/engine/reference/builder/#copy).
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:205:80: E501 line too long (82 > 79 characters)
        """Checks to make sure the sysroot directory is setup correctly.

        Raises FileNotFoundError's if any of the components necessary for cross compilation is
        missing. Copies the Dockerfile_workspace to the 'sysroot' directory in order to copy the
        assets to it (see https://docs.docker.com/engine/reference/builder/#copy).
        """
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:215:80: E501 line too long (97 > 79 characters)
                raise FileNotFoundError(QEMU_DIR_ERROR_STRING.substitute(qemu_dir=QEMU_DIR_NAME))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:217:80: E501 line too long (99 > 79 characters)
                raise FileNotFoundError(QEMU_EMPTY_ERROR_STRING.substitute(qemu_dir=QEMU_DIR_NAME))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:219:80: E501 line too long (83 > 79 characters)
            shutil.copy(str(self._dockerfile_directory), str(self._target_sysroot))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:229:80: E501 line too long (93 > 79 characters)
        logger.info('Fetching sysroot base image: {}'.format(self._docker_config.base_image))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:239:22: Q000 Remove bad quotes
        logger.debug("Build Arguments: {}")
                     ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:257:80: E501 line too long (81 > 79 characters)
            # There are usually two outputs we want to capture, stream and error.
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:262:80: E501 line too long (84 > 79 characters)
                logger.exception('Error building sysroot image. The following error'
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:264:80: E501 line too long (87 > 79 characters)
                raise docker.errors.BuildError(reason=error_line, build_log=error_line)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:270:80: E501 line too long (86 > 79 characters)
        logger.info('Successfully created sysroot docker image: {}'.format(image_tag))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:274:80: E501 line too long (82 > 79 characters)
        logger.info('Exporting sysroot to path [{}]'.format(self._target_sysroot))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:278:80: E501 line too long (82 > 79 characters)
        sysroot_tarball_path = Path(tmp_sysroot_dir) / (SYSROOT_DIR_NAME + '.tar')
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:284:80: E501 line too long (90 > 79 characters)
            sysroot_container = DOCKER_CLIENT.containers.run(image=image_tag, detach=True)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:289:80: E501 line too long (92 > 79 characters)
                relevant_dirs = ['lib', 'usr', 'etc', 'opt', 'root_path', 'ros2_ws/install']
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:292:80: E501 line too long (94 > 79 characters)
                    if re.match('^({}).*'.format('|'.join(relevant_dirs)), m.name) is not None
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:294:80: E501 line too long (91 > 79 characters)
                sysroot_tar.extractall(str(self._target_sysroot), members=relevant_members)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:298:80: E501 line too long (90 > 79 characters)
        logger.info('Success exporting sysroot to path [{}]'.format(self._target_sysroot))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:308:80: E501 line too long (80 > 79 characters)
                cc_root_dir=self._target_sysroot, distro=self._platform.distro))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:315:80: E501 line too long (83 > 79 characters)
        cc_system_setup_script_path = self._target_sysroot / 'cc_system_setup.bash'
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:316:80: E501 line too long (87 > 79 characters)
        cc_system_setup_script_contents = CC_BUILD_SYSTEM_SETUP_SCRIPT_TEMPLATE.render(
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:331:80: E501 line too long (81 > 79 characters)
        logger.info("Executing 'bash {}'".format(self._system_setup_script_path))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:333:80: E501 line too long (82 > 79 characters)
        logger.info("Executing 'source {}'".format(self._build_setup_script_path))
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:334:80: E501 line too long (82 > 79 characters)
        subprocess.run(['source', str(self._build_setup_script_path)], shell=True)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py:336:1: D102 Missing docstring in public method
    def execute_cc_pipeline(self) -> bool:
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:15:80: E501 line too long (116 > 79 characters)
CC_COMPLETE_STRING = Template('To setup the cross compilation build environment: 1. Run the command below to setup '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:16:31: Q003 Change outer quotes to avoid escaping inner quotes
                              'using sysroot\'s GLIBC for cross-compilation.\n`bash $system_setup_script_path`\n2. Run '
                              ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:16:80: E501 line too long (120 > 79 characters)
                              'using sysroot\'s GLIBC for cross-compilation.\n`bash $system_setup_script_path`\n2. Run '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:17:80: E501 line too long (117 > 79 characters)
                              'the command below to export the environment variables used by the cross-compiled ROS '
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:23:20: Q001 Remove bad quotes from multiline string
    example_text = '''Example usage:
                   ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:24:1: W293 blank line contains whitespace
    example_text = '''Example usage:
    
    python3 create_cc_sysroot.py --arch armhf --os debian
    python3 create_cc_sysroot.py --arch aarch64 --os ubuntu
    python3 create_cc_sysroot.py -a armhf -o ubuntu -d dashing -r fastrtps \
--sysroot-base-image arm64v8/ubuntu:bionic
    '''
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:92:80: E501 line too long (90 > 79 characters)
        help="The full path to the directory containing 'sysroot'. The 'ros2_ws/src' and "
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:93:80: E501 line too long (90 > 79 characters)
             "'qemu-user-static' directories and the 'Dockerfile_workspace' file used to "
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:94:14: Q000 Remove bad quotes
             "cross-compile the ROS packages should all be in this directory. Defaults to the "
             ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:94:80: E501 line too long (95 > 79 characters)
             "cross-compile the ROS packages should all be in this directory. Defaults to the "
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:95:14: Q000 Remove bad quotes
             "current working directory.")
             ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:108:80: E501 line too long (107 > 79 characters)
    sysroot_create = SysrootCompiler(cc_root_dir=args.sysroot_path, ros_workspace_dir=args.sros2_workspace,
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:109:80: E501 line too long (82 > 79 characters)
                                     platform=platform, docker_config=docker_args)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:25:1: I100 Import statements are in the wrong order. 'from cross_compile.sysroot_compiler import DockerConfig' should be before 'import pytest' and in a different group.
from cross_compile.sysroot_compiler import DockerConfig
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:26:1: I100 Import statements are in the wrong order. 'from cross_compile.sysroot_compiler import DOCKER_WS_NAME' should be before 'from cross_compile.sysroot_compiler import DockerConfig'
from cross_compile.sysroot_compiler import DOCKER_WS_NAME
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:30:1: I100 Import statements are in the wrong order. 'from cross_compile.sysroot_compiler import SYSROOT_DIR_NAME' should be before 'from cross_compile.sysroot_compiler import SysrootCompiler'
from cross_compile.sysroot_compiler import SYSROOT_DIR_NAME
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:61:1: D401 First line should be in imperative mood
    """Creates mock directories to correctly construct the SysrootCreator."""
^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:86:80: E501 line too long (80 > 79 characters)
    test_config_string = 'Base Image: {}\nNetwork Mode: {}\nCaching: {}'.format(
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:87:80: E501 line too long (80 > 79 characters)
        args.sysroot_base_image, args.docker_network_mode, args.sysroot_nocache)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:93:80: E501 line too long (80 > 79 characters)
def test_sysroot_compiler_constructor(platform_config, docker_config, tmp_path):
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:98:80: E501 line too long (81 > 79 characters)
        str(sysroot_dir), str(ros_workspace_dir), platform_config, docker_config)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:108:80: E501 line too long (81 > 79 characters)
        str(sysroot_dir), str(ros_workspace_dir), platform_config, docker_config)
                                                                               ^
/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/test/test_sysroot_compiler.py:119:80: E501 line too long (81 > 79 characters)
        str(sysroot_dir), str(ros_workspace_dir), platform_config, docker_config)
                                                                               ^
flake8 reported 84 errors
________________________ test_write_cc_build_setup_file ________________________
test/test_sysroot_compiler.py:109: in test_write_cc_build_setup_file
    setup_script_path = sysroot_compiler._write_cc_build_setup_script()
cross_compile/sysroot_compiler.py:309: in _write_cc_build_setup_script
    with open(str(cc_build_setup_file_path), 'w') as out_f:
E   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmoulard/pytest-of-tmoulard/pytest-5/test_write_cc_build_setup_file0/sysroot/sysroot/cc_build_setup.bash'
----------------------------- Captured stderr call -----------------------------
Sysroot directory not found at '/tmp/tmoulard/pytest-of-tmoulard/pytest-5/test_write_cc_build_setup_file0/sysroot'. Make sure you specify the full path to the directory containing 'sysroot'.
Traceback (most recent call last):
  File "/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py", line 188, in __init__
    self._setup_sysroot_dir()
  File "/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py", line 225, in _setup_sysroot_dir
    sysroot_dir=self._cc_root_dir))
FileNotFoundError: Sysroot directory not found at '/tmp/tmoulard/pytest-of-tmoulard/pytest-5/test_write_cc_build_setup_file0/sysroot'. Make sure you specify the full path to the directory containing 'sysroot'.
_______________________ test_write_cc_system_setup_file ________________________
test/test_sysroot_compiler.py:120: in test_write_cc_system_setup_file
    setup_script_path = sysroot_compiler._write_cc_system_setup_script()
cross_compile/sysroot_compiler.py:319: in _write_cc_system_setup_script
    with open(str(cc_system_setup_script_path), 'w') as out_f:
E   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmoulard/pytest-of-tmoulard/pytest-5/test_write_cc_system_setup_fil0/sysroot/sysroot/cc_system_setup.bash'
----------------------------- Captured stderr call -----------------------------
Sysroot directory not found at '/tmp/tmoulard/pytest-of-tmoulard/pytest-5/test_write_cc_system_setup_fil0/sysroot'. Make sure you specify the full path to the directory containing 'sysroot'.
Traceback (most recent call last):
  File "/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py", line 188, in __init__
    self._setup_sysroot_dir()
  File "/home/ANT.AMAZON.COM/tmoulard/ros2_ws/src/cross_compile/cross_compile/sysroot_compiler.py", line 225, in _setup_sysroot_dir
    sysroot_dir=self._cc_root_dir))
FileNotFoundError: Sysroot directory not found at '/tmp/tmoulard/pytest-of-tmoulard/pytest-5/test_write_cc_system_setup_fil0/sysroot'. Make sure you specify the full path to the directory containing 'sysroot'.
- generated xml file: /home/ANT.AMAZON.COM/tmoulard/ros2_ws/build/cross_compile/pytest.xml -

----------- coverage: platform linux, python 3.6.8-final-0 -----------
Name    Stmts   Miss  Cover   Missing
-------------------------------------

========================= 3 failed, 3 passed in 0.94s ==========================
Coverage.py warning: Module b9_ros_github_tools was never imported. (module-not-imported)
Coverage.py warning: No data was collected. (no-data-collected)
/home/ANT.AMAZON.COM/tmoulard/.local/lib/python3.6/site-packages/pytest_cov/plugin.py:254: PytestWarning: Failed to generate report: No data to report.

  self.cov_controller.finish()
--- stderr: cross_compile
Coverage.py warning: Module b9_ros_github_tools was never imported. (module-not-imported)
Coverage.py warning: No data was collected. (no-data-collected)
/home/ANT.AMAZON.COM/tmoulard/.local/lib/python3.6/site-packages/pytest_cov/plugin.py:254: PytestWarning: Failed to generate report: No data to report.

  self.cov_controller.finish()
---
Finished <<< cross_compile [2.56s]	[ with test failures ]

Summary: 1 package finished [3.76s]
  1 package had stderr output: cross_compile
  1 package had test failures: cross_compile
```

[test.log](https://github.com/aws-ros-dev/cross_compile/files/3692795/test.log)

**Comment by piraka9011** [comment]
Fixed by #15, #17

**Comment by thomas-moulard** [comment]
```
test/test_flake8.py::test_flake8 FAILED                                  [ 16%]
test/test_sysroot_compiler.py::test_get_workspace_image_tag PASSED       [ 33%]
test/test_sysroot_compiler.py::test_docker_config_args PASSED            [ 50%]
test/test_sysroot_compiler.py::test_sysroot_compiler_constructor PASSED  [ 66%]
test/test_sysroot_compiler.py::test_write_cc_build_setup_file FAILED     [ 83%]
test/test_sysroot_compiler.py::test_write_cc_system_setup_file FAILED    [100%]
```

**Comment by piraka9011** [comment]
See #19

**Comment by thomas-moulard** [comment]
```
test/test_flake8.py::test_flake8 FAILED                                  [ 16%]                                          
test/test_sysroot_compiler.py::test_get_workspace_image_tag PASSED       [ 33%]                                                           
test/test_sysroot_compiler.py::test_docker_config_args PASSED            [ 50%]                                                          
test/test_sysroot_compiler.py::test_sysroot_compiler_constructor PASSED  [ 66%]                    
test/test_sysroot_compiler.py::test_write_cc_build_setup_file FAILED     [ 83%]                                                                                                                                              
test/test_sysroot_compiler.py::test_write_cc_system_setup_file FAILED    [100%]
```

**Comment by piraka9011** [comment]
Fixed by #21
